src/ceph.in

```diff
@@ -552,8 +552,8 @@ def ping_monitor(cluster_handle, name, timeout):
         return 1
 
     mon_id = name[len('mon.'):]
+    run_in_thread(cluster_handle.connect, timeout=timeout)
```

Comment:
> Perhaps the right fix is in build_initial_monmap, then, to actually use mon_initial_members and/or the DNS lookup it does

@dmick, "mon_host", "mon.*" and DNS SRV are used to build the monmap when "mkfs", and the  founded monitors are persisted into the monmap. while "mon_initial_members" is used when the monitor joins the cluster for the first time. and we need the majority in "mon_initial_members" to form the quorum. but the monitors not listed are still probed and welcomed to join the quorum.

so i'd say that the monitor collected in "mkfs" is a set not necessarily identical to the one defined by "mon_initial_members". and they serve different purposes. we could **try** to use the "mon_initial_members" when building the initial map when mkfs, but we would need to do the dedup to remove the duplicated monitor included in both sets. that would incur more complexity.

> I can table the mon ping fix and just get the help fix in

maybe we can also document this in http://docs.ceph.com/docs/master/rados/configuration/mon-config-ref/#initial-members a little bit?